extensions/wikia/ImageReview/ImageListGetter.class.php

```diff
@@ -0,0 +1,384 @@
+<?php
+
+use Wikia\Logger\WikiaLogger;
+
+class ImageListGetter extends WikiaModel {
+
+	/**
+	 * LIMIT_IMAGES_FROM_DB should be a little greater than LIMIT_IMAGES, so if
+	 * we fetch a few icons from DB, we can skip them
+	 */
+	const LIMIT_IMAGES = 20;
+	const LIMIT_IMAGES_FROM_DB = 128;
+	const IMAGE_REVIEW_THUMBNAIL_SIZE = 250;
+
+	const FLAG_SUSPICOUS_USER = 2;
+	const FLAG_SUSPICOUS_WIKI = 4;
+	const FLAG_SKIN_DETECTED = 8;
+
+	private $userId = 0;
+	private $timestamp;
+	private $state;
+	private $order;
+	private $imageStateUpdater;
+	private $imageList = [];
+
+	function __construct( int $timestamp, int $state, int $order ) {
+		parent::__construct();
+		$user = $this->wg->User;
+		$this->userId = $user->getId();
+		$this->timestamp = wfTimestamp( TS_DB, $timestamp );
+		$this->state = $state;
+		$this->order = ( new ImageReviewOrderGetter() )->getOrder( $user, $order );
+		$this->imageStateUpdater= new ImageStateUpdater();
+	}
+
+	/**
+	 * Get batch of images for Image Review Tool.
+	 *
+	 * The goal is to return up to 20 images to the client for review. In order to
+	 * do that, we trying fetching images in 3 different ways, combining any images
+	 * we've found along the way. The ways are (in order):
+	 *
+	 * 1.) Check if the given timestamp matches a previously reviewed batch of images.
+	 * 2.) Check if there are any outstanding reviews (reviews that this reviewer started
+	 *     but hasn''t finish by making a decision on the images).
+	 * 3.) Create a new review
+	 */
+	public function getImageList() {
+		$this->fetchPreviousReviewFromTimestamp();
+
+		if ( !$this->fetchedEnoughImages() ) {
+			$this->fetchUnfinishedReview();
+		}
+
+		if ( !$this->fetchedEnoughImages() ) {
+			$this->fetchNewReview();
+		}
+
+		return $this->imageList;
+	}
+
+	private function fetchPreviousReviewFromTimestamp() {
+		$this->fetchAndPrepareImageResults( function() {
+			return ( new WikiaSQL() )
+				->SELECT_ALL()
+				->FROM( 'image_review' )
+				->WHERE( 'reviewer_id' )->EQUAL_TO( $this->userId )
+				->AND_( 'review_start' )->EQUAL_TO( $this->timestamp )
+				->ORDER_BY( ImageReviewOrderGetter::PRIORITY_LATEST_SQL )
+				->LIMIT( self::LIMIT_IMAGES_FROM_DB )
+				->runLoop( $this->getDatawareDB(), function ( &$images, $row ) {
+					$images[] = $row;
+				});
+			},
+			false
+		);
+	}
+
+	private function fetchUnfinishedReview() {
+		$this->fetchAndPrepareImageResults( function() {
+			return ( new WikiaSQL() )
+				->SELECT_ALL()
+				->FROM( 'image_review' )
+				->AND_( 'reviewer_id' )->EQUAL_TO( $this->userId )
+				->AND_( 'state' )->EQUAL_TO( $this->getNewState() )
```

Comment:
![MINOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png) Define a constant instead of duplicating this literal "state" 6 times. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS1192)